zad2/warriors.rb

```diff
@@ -1,12 +1,75 @@
-class Warrior
-  def initialize(name:, level:)
+class Character
+  attr_accessor :name, :level
+  def initialize(name: , level:)
+    @name = name
+    @level = valid_level(level)
   end
 
   def strength
+    @level + [*1..12].sample
+  end
+  
+  #we make sure that level is in range 1..99
+  def valid_level(lvl)
+    lvl.between?(1,99) ? lvl : (lvl > 99 ? 99 : 1)
+  end
+
+  def card
+    puts("#{@name}, lvl #{@level}")
+  end
+
+end
+
+class Warrior < Character
+  def points(p)
+    #level increases only if the opponent's level was lower
+    if p > 0 
```

Comment:
w takich przypadkach polecam tzn guard clause
https://www.thechrisoshow.com/2009/02/16/using-guard-clauses-in-your-ruby-code/